src/modules/viewmodel_sway.rs

```diff
@@ -0,0 +1,119 @@
+//! Viewmodel sway
+use std::f32::consts::PI;
+
+use super::Module;
+use crate::hooks::engine::{self, ref_params_s};
+use crate::modules::cvars::CVar;
+use crate::utils::*;
+
+pub struct ViewmodelSway;
+impl Module for ViewmodelSway {
+    fn name(&self) -> &'static str {
+        "Viewmodel sway"
+    }
+
+    fn description(&self) -> &'static str {
+        "Viewmodel sway."
+    }
+
+    fn cvars(&self) -> &'static [&'static CVar] {
+        static CVARS: &[&CVar] = &[
+            &BXT_VIEWMODEL_SWAY,
+            &BXT_VIEWMODEL_SWAY_MAX,
+            &BXT_VIEWMODEL_SWAY_RATE,
+        ];
+        CVARS
+    }
+
+    fn is_enabled(&self, marker: MainThreadMarker) -> bool {
+        engine::ClientDLL_CalcRefdef.is_set(marker) && engine::cl_viewent.is_set(marker)
+    }
+}
+
+static BXT_VIEWMODEL_SWAY: CVar = CVar::new(
+    b"bxt_viewmodel_sway\0",
+    b"0\0",
+    "\
+Setting to `1` enables weapon sway. Default is `100` max and `0.02` rate.",
+);
+
+static BXT_VIEWMODEL_SWAY_MAX: CVar = CVar::new(
+    b"bxt_viewmodel_sway_max\0",
+    b"100\0",
+    "\
+Max of sway.",
+);
+
+static BXT_VIEWMODEL_SWAY_RATE: CVar = CVar::new(
+    b"bxt_viewmodel_sway_rate\0",
+    b"0.02\0",
+    "\
+Rate of sway.",
+);
+
+static LAST_VIEWANGLES: MainThreadCell<glam::Vec3> = MainThreadCell::new(glam::vec3(0., 0., 0.));
+
+fn normalize_angle(angle: f32) -> f32 {
+    let new_angle = angle % 360.;
+
+    if new_angle > 180. {
+        new_angle - 360.
+    } else if new_angle < -180. {
+        new_angle + 360.
+    } else {
+        new_angle
+    }
+}
+
+// credits to https://github.com/edgarbarney/halflife-planckepoch/commit/9a38b22acf97a6d5065b466c060e6bcd10c6db2c
+fn smooth_values(start: f32, end: f32, speed: f32) -> f32 {
+    let d = end - start;
+    let dabs = d.abs();
+
+    if dabs > 0.01 {
+        if (end - start) > 0. {
+            start + (dabs * speed)
+        } else {
+            start - (dabs * speed)
+        }
+    } else {
+        end
+    }
+}
+
+pub fn with_viewmodel_sway(marker: MainThreadMarker, rp: &ref_params_s) {
+    if !BXT_VIEWMODEL_SWAY.as_bool(marker) {
+        return;
+    }
+
+    if rp.paused == 1 {
+        return;
+    }
+
+    let view = unsafe { &mut *engine::cl_viewent_viewmodel.get(marker) };
```

Comment:
Add a safety comment saying that no engine functions are called while the reference is active. You can search for existing SAFETY comments like this and copy the text.